apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Synchronize/02-QRCodeStep.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { Flex, Icons, NumberedList, Text } from "@ledgerhq/react-ui";
+import styled, { useTheme } from "styled-components";
+
+type Props = {
+  displayPinCode: () => void;
+};
+export default function SynchWithQRCodeStep({ displayPinCode }: Props) {
+  const { t } = useTranslation();
+
+  const { colors } = useTheme();
+
+  const steps = [
+    { element: t("walletSync.synchronize.qrCode.steps.step1") },
+    { element: t("walletSync.synchronize.qrCode.steps.step2") },
+    { element: t("walletSync.synchronize.qrCode.steps.step3") },
+  ];
+
+  // TO CHANGE WHEN INTRAGRATION WITH TRUSTCHAIN
```

Comment:
notes to anticipate a bit the integration:

this code may just disappear because we need to be able to "control" the steps, it's not the step that will do a call so some api and do something based on this because we will have a global function to call at the beginning and we will just hook to the various events.

so i think we don't need to have the "displayPinCode" injected as a prop neither, it's going to be a switch of step from the parent directly.

see the usage here for instance: https://github.com/LedgerHQ/ledger-live/blob/live-12627/apps/web-tools/trustchain/components/App.tsx#L468-L479 (generally speaking, this App.tsx file will act as general/canonical examples for all trustchain integrations)

```ts
createQRCodeHostInstance({
      onDisplayQRCode: url => {
        setUrl(url);
      },
      onDisplayDigits: digits => {
        setDigits(digits);
      },
      addMember: async member => {
        // this sdk will be somehow available as a react context when we do the trustchain integration
        const jwt = await sdk.liveAuthenticate(trustchain, liveCredentials);
        await sdk.addMember(jwt, trustchain, liveCredentials, member);
      },
    })
      .catch(e => {
        if (e instanceof InvalidDigitsError) {
          return; // in this case, we could display an error of invalid digital error (it can also be handled by the generic component)
        }
        setError(e); // in this case, we render the generic error (we have a generic component for handling of errors) , typically it can be Networking error, but there could be also protocol errors
      })
      .then(() => {
        // everything is finished. the other member was added ✅ 
      });
```

Reply:
Yeah nice, I just putted that to simulate a behavior to develop UI but I'll make another PR right now to start the job on this part :)

Reply:
We are currently able to control which step we want to display :)